MetaMask/metamask-extension
# UX: Multichain: Fix Network Picker when Test Networks Disabled

## Explanation

I noticed that when I was on Goerli network, and I used the Network Picker to turn off the Test Networks setting, the network picker would no longer show the name "Goerli" -- it showed no name at all, which is a bug.  It was happening because `getAllNetworks` only returns test network names when `showTestNetworks` is on.  This PR:

1.  Makes `getAllNetworks` always return all networks, regardless of any setting
2. Introduces `getAllEnabledNetworks` to return only enabled networks (i.e. hides test networks when the setting is off)

## Manual Testing Steps

1.  Switch to Georli
2. Open the Network Picker
3. Turn off test networks
4. See the Goerli name still in the Network Picker

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
